pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -1050,9 +1050,13 @@ protected void handleConnect(CommandConnect connect) {
                         .getAuthenticationProvider(originalAuthMethod);
 
                 if (originalAuthenticationProvider == null) {
-                    throw new AuthenticationException(
-                            String.format("Can't find AuthenticationProvider for original role"
-                                    + " using auth method [%s] is not available", originalAuthMethod));
+                    authRole = getBrokerService().getAuthenticationService().getAnonymousUserRole()
```

Comment:
Please add a comment to clarify the condition that has `connect.hasOriginalPrincipal()` but no `originalAuthMethod` means the proxy wants the broker to use anonymousUserRole. :)

Reply:
Fixed